src/cf.data.pre

```diff
@@ -7112,11 +7112,28 @@ LOC: Config.visibleHostname
 DEFAULT: none
 DEFAULT_DOC: Automatically detect the system host name
 DOC_START
-	If you want to present a special hostname in error messages, etc,
-	define this.  Otherwise, the return value of gethostname()
-	will be used. If you have multiple caches in a cluster and
-	get errors about IP-forwarding you must set them to have individual
-	names with this setting.
+	The hostname of this proxy.
+
+	Hosts connected to the Internet or handling messages which travel
+	over the Internet are required by RFC 1123 to have a name which
+	can be resolved in DNS (an FQDN).
+
+	If not set, auto-detection of the system host name will be performed
+	based on the following details (in this order):
+		- rDNS name for the first listening port address
+		- the name returned by gethostname(2)
+		- the name "localhost"
+
+	This name will be used in generated responses sent to clients
```

Comment:
This sentence is very difficult to understand. `objects they need to display the proxy name` just does not compute. I suggest something like `This directive is used for naming this proxy in generated responses sent to clients. For example, it is used to expand %h in a Squid error page template`. Please use the correct `%code` if you follow this sketch -- I did not check whether `%h` is correct.

Reply:
You want all of the past present and future codes to be listed as well as all the non-errorpage uses? error pages naming the proxy is just the most obvious and best-known example.